isisd/isis_nb_config.c

```diff
@@ -2436,6 +2436,8 @@ int isis_instance_segment_routing_prefix_sid_map_prefix_sid_create(
 	pcfg = isis_sr_cfg_prefix_add(area, &prefix, SR_ALGORITHM_SPF);
 	nb_running_set_entry(args->dnode, pcfg);
 
+	lsp_regenerate_schedule(area, area->is_type, 0);
```

Comment:
`lsp_regenerate_schedule` is already called in `isis_instance_segment_routing_prefix_sid_map_prefix_sid_apply_finish`, which is executed after `isis_instance_segment_routing_prefix_sid_map_prefix_sid_create`.

I see no reason to call `lsp_regenerate` here.

---

Try to enable `debug isis lsp-sched` and check the log. You should see the LSP regeneration just after executing `segment-routing prefix ...`.

```
2024/08/20 03:16:44.320877 ISIS: [M7Q4P-46WDR] vty[5]@(config-router)# segment-routing prefix 2001:db8:1000::1/128 index 1000
2024/08/20 03:16:44.337220 ISIS: [KE21Z-HRMJH] ISIS (1): Scheduling regeneration of L1 LSPs, not including PSNs Caller: isis_instance_segment_routing_prefix_sid_map_prefix_sid_apply_finish isisd/isis_nb_config.c:2561
2024/08/20 03:16:44.337234 ISIS: [WW5PN-1HJS3] ISIS (1): Checking whether L1 needs to be scheduled
2024/08/20 03:16:44.337239 ISIS: [VRV8G-PW3ER] ISIS (1): Will schedule regen timer. Last run was: 1724116598, Now is: 1724116604
2024/08/20 03:16:44.337255 ISIS: [JNWST-M6Q1H] ISIS (1): Last generation was more than lsp_gen_interval ago. Scheduling for execution in 99 ms due to the instability timer.
2024/08/20 03:16:44.439357 ISIS: [ZCEC7-S9PJZ] ISIS (1): LSP L1 refresh timer expired. Refreshing LSP...
2024/08/20 03:16:44.439631 ISIS: [KR60V-W9SMF] ISIS (1): Rebuilt L1 LSP. Set triggered regenerate to non-pending.

...

2024/08/20 10:58:38.755278 ISIS: [M7Q4P-46WDR] vty[5]@(config-router)# no segment-routing prefix 2001:db8:1000::1/128
2024/08/20 10:58:38.842525 ISIS: [KE21Z-HRMJH] ISIS (1): Scheduling regeneration of L1 LSPs, not including PSNs Caller: isis_instance_segment_routing_prefix_sid_map_prefix_sid_destroy isisd/isis_nb_config.c:2458
2024/08/20 10:58:38.842729 ISIS: [WW5PN-1HJS3] ISIS (1): Checking whether L1 needs to be scheduled
2024/08/20 10:58:38.842731 ISIS: [VRV8G-PW3ER] ISIS (1): Will schedule regen timer. Last run was: 1724143984, Now is: 1724144318
2024/08/20 10:58:38.842747 ISIS: [JNWST-M6Q1H] ISIS (1): Last generation was more than lsp_gen_interval ago. Scheduling for execution in 99 ms due to the instability timer.
2024/08/20 10:58:38.943123 ISIS: [ZCEC7-S9PJZ] ISIS (1): LSP L1 refresh timer expired. Refreshing LSP...
2024/08/20 10:58:38.945078 ISIS: [KR60V-W9SMF] ISIS (1): Rebuilt L1 LSP. Set triggered regenerate to non-pending.
```

If you don't see any LSP regeneration message, please share your FRR configuration.